File: LeetCode/1015. Smallest Integer Divisible by K.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    int smallestRepunitDivByK(int k) {
        int v[100000] = {0}, n = 0, ans = 0;
        while (true) 
        {   
            ans++ ;
            n = (n * 10 + 1) % k; 
            if (!n) 
                return ans;
            if (v[n]) 
                return -1 ;
            v[n] = 1 ;
        }
    }
};

File: LeetCode/1015. Smallest Integer Divisible by K.java
/***  KNOWLEDGE IS POWER  ***/

class Solution {
    public int smallestRepunitDivByK(int k) {
        int[] hit = new int[k];
        int n = 0, ans = 0;
        while (true) 
        {
            ans++ ;
            n = (n * 10 + 1) % k; 
            if (n == 0) 
                return ans ;
            if (hit[n] == 1) 
                return -1;
            hit[n] = 1 ;
        }
    }
}
